js/parts/DataGrouping.js

```diff
@@ -505,7 +505,20 @@ seriesProto.groupData = function (xData, yData, groupPositions, approximation) {
 
     // Start with the first point within the X axis range (#2696)
     for (i = 0; i <= dataLength; i++) {
-        if (xData[i] >= groupPositions[0]) {
+        var diff = xData[i] - groupPositions[0];
+        if (diff >= 0) {
+            break;
+        } else if (
+            groupPositions.info.unitName === 'week' &&
+            diff > -groupPositions.info.unitRange / 7
+        ) {
+            // Add an extra position before first position, becasue
+            // getTimeTicks trims first position if it starts on Sunday (#7051)
```

Comment:
Isn't this a bug in getTimeTicks? Shouldn't it extend the tick positions to previous Monday instead of trimming the first position? In that case we must patch `getTimeTicks` instead of creating a workaround in DataGrouping.js.

Proposal for a unit test: http://jsfiddle.net/highcharts/1v9kbpy7/

In this case we pass Sunday as the min date, yet the first tick is placed on the next Monday.

Reply:
Yes, this is actually bug in getTimeTicks, but it only applies to dataGrouping so I decided for a workaround in dataGrouping. I couldn't imagine a case where we need this fixed for time ticks.

Reply:
I think this is the case: http://jsfiddle.net/highcharts/zsywLdug/ . The starting point is on Sunday 12:00.

If we change the starting point to Saturday 12:00, the axis will start on the Monday before, which is the correct behaviour: http://jsfiddle.net/highcharts/zsywLdug/2/

Reply:
Ah, `startOnTick`. Let me create a separate branch and a fix for it.